game/scripts/vscripts/components/points/points.lua

```diff
@@ -17,5 +24,81 @@ function PointsManager:Init ()
   end
 
   -- set initial values for current scores
-  CustomNetTables:SetTableValue( "team_scores", "score", { radiant = 0, dire = 0 } )
+  CustomNetTables:SetTableValue( "team_scores", "score", { goodguys = 0,
+                                                           badguys = 0
+                                                         })
+
+  -- Start Thinking
+  Timers:CreateTimer(0, Dynamic_Wrap(PointsManager, "Think"))
+end
+
+function PointsManager:Think ()
+  DebugPrint("[points/PointsManager] Thinking..")
+
+  local interval = 5 -- when do we want to check for win conditions
+  local limit = CustomNetTables:GetTableValue("team_scores", "limit").value
+  local scores = CustomNetTables:GetTableValue("team_scores", "score")
+  local goodguys = scores.goodguys
+  local badguys = scores.badguys
+
+  DebugPrintTable(limit)
+  DebugPrintTable(scores)
+  DebugPrint("haveGoodguysWon: " .. tostring(PointsManager.haveGoodguysWon))
+  DebugPrint("haveBadguysWon: " .. tostring(PointsManager.haveBadguysWon))
+
+  PointsManager:SetPoints(PointsManager.goodguysName, 200)
+
+  if PointsManager.haveGoodguysWon or PointsManager.haveBadguysWon then
+    return
+  end
+
+  if goodguys >= limit then
+    PointsManager:onTeamWin(PointsManager.goodguysName)
+  elseif badguys >= limit then
+    PointsManager:onTeamWin(PointsManager.badguysName)
+  end
+
+  return interval
+end
+
+function PointsManager:onTeamWin (side)
+  -- NOTE Maybe set new GameState
+  --GameRules.SetMode(DOTA_GAMERULES_STATE_POST_GAME)
+  DebugPrint("[points/PointsManager] " .. side .. " wins!")
+
+  if side == PointsManager.goodguysName then
+    PointsManager.haveGoodguysWon = true
+    GameRules:SetGameWinner(PointsManager.goodguysID)
+  elseif side == PointsManager.badguysName then
+    PointsManager.haveBadguysWon = true
+    GameRules:SetGameWinner(PointsManager.badguysID)
+  end
+end
+
+function PointsManager:SetPoints (side, newPoints)
+  --DebugPrint("[points/PointsManager] Set Score of " .. side .. " to " .. newScore .. ".")
+
+  local score = CustomNetTables:GetTableValue("team_scores", "score")
+
+  if side == PointsManager.goodguysName then
+    score.goodguys = newPoints
+  elseif side == PointsManager.badguysName then
+    score.badguys = newPoints
+  end
+
+  CustomNetTables:SetTableValue("team_scores", "score", score)
+end
+
+function PointsManager:IncrementPoints (side)
```

Comment:
Can this take in an optional parameter, `amount` which specifies how much to add? also `AddPoints` is probably more intuitive
```lua
function PointsManager:AddPoints (side, amount)
  if amount == nil then
    amount = 1
  end

  -- then, later, add amount instead of 1
```